src/main/scala/co/teapot/tempest/server/TempestDatabaseClient.scala

```diff
@@ -185,19 +182,21 @@ class TempestSQLDatabaseClient(config: DatabaseConfig) extends TempestDatabaseCl
       if (nodeIds.isEmpty)
         return Map.empty
       validateAttributeName(attributeName)
-      for (id <- nodeIds) {
-        validateId(id)
-      }
 
-      val idList = stringsToPostgresSet(nodeIds)
+
+      // generates (?, ?, ..., ?), one question mark per node id
+      val sqlInParams = Iterator.fill(nodeIds.size)("?").mkString("(", ",", ")")
```

Comment:
I haven't check but quick googling vaguely suggests that postgres will create a temporary table and perform an inner join if it sees many values in `IN` clause. The jdbc seems to be limited at 2^16 parameters.